src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,164 @@
+
+        task_info_file = self.get_task_info_file()
+        df = pd.read_json(task_info_file, lines=True)
+
+        tasks_df = df[(df["kbase_collection"] == self.kbase_collection) &
+                      (df["load_ver"] == self.load_ver) &
+                      (df["tool"] == self.tool)]
+
+        tasks_df.sort_values(by="job_start_time", ascending=False, inplace=True)
+
+        return tasks_df
+
+    def get_latest_task(self):
+        """
+        Get the latest task information
+
+        Tasks are sorted by job_start_time in descending order. The latest task is the first row.
+        """
+        if not self.task_exists:
+            return {}
+
+        return self.tasks_df.iloc[0].to_dict()
+
+    def get_job_status_fm_nersc(self, job_id):
+        """
+        Get the job status from NERSC using squeue command.
+
+        When job is queued, squeue command will return code 0 and out put will be like (with no job info just the header):
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+
+        When job is running, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        5987779 regular_m submit_t      tgu  R       1:20      2 nid[005689-005690]
+
+        When job is finished/failed:
+            * job status is cached for a period of time (about 10 mins). squeue command will return code 0 and
+            out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+            * after the period of time, squeue command will return error code 1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+        """
+
+        std_out_file, std_err_file, exit_code = tf_common.run_nersc_command(['squeue', '-j', str(job_id)],
+                                                                            self.job_dir, log_file_prefix='squeue',
+                                                                            check_return_code=False)
+
+        if exit_code != 0:
+            # job (successfully) finished status is checked by checking the `done.tasks.txt.tfin` file in job directory
+            return JobStatus.FAILED
+
+        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
+            squeue_out = std_out.read().strip()
+            if str(job_id) in squeue_out:
+                return JobStatus.RUNNING
+            else:
+                return JobStatus.PENDING
+
+    def retrieve_latest_task_status(self):
+        """
+        Retrieve the latest task status.
+
+        """
+        if not self.task_exists:
+            raise ValueError(f"Task does not exist for {self.kbase_collection}, {self.load_ver}, {self.tool}")
+
+        latest_task = self.get_latest_task()
+
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
```

Comment:
How does this file get created?

Reply:
once the task finishes, it will be generated in the job_dir by nersc slurm.

Reply:
Same comment as https://github.com/kbase/collections/pull/177#discussion_r1134487789

Reply:
👍

Reply:
:+1: